### PR TITLE
feat: handle CERTIFICATE security token type

### PR DESCRIPTION
**Pull request by ytvnr**
## Issue
https://gravitee.atlassian.net/browse/APIM-3985

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-jbjfcekmtr.chromatic.com)

### CODE REVIEW

File: gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-core/src/main/java/io/gravitee/gateway/security/core/SubscriptionTrustStoreLoader.java [hunk 1]
```diff
@@ -21,23 +21,37 @@
...
         super(SubscriptionTrustStoreLoaderOption.empty());
-        this.subscription = subscription;
         this.id = "subscription_cert_%s".formatted(subscription.getId());
+        final String pem = new String(Base64.getDecoder().decode(subscription.getClientCertificate()));
+        keystore = KeyStoreUtils.initFromPemCertificate(pem, getPassword(), subscription.getId());
+
+        try {
+            digest = DigestUtils.md5DigestAsHex(keystore.getCertificate(subscription.getId()).getEncoded());
+        } catch (CertificateEncodingException | KeyStoreException e) {
+            log.error("An error occurred while computing digest for Subscription certificate {}", subscription.getId(), e);
```
Comment (dev-marek) [inline gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-core/src/main/java/io/gravitee/gateway/security/core/SubscriptionTrustStoreLoader.java]: Is it ok to just print error and move on?

**Review by dev-marek — APPROVED** [review]
LGTM with one question

File: gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-core/src/main/java/io/gravitee/gateway/security/core/SubscriptionTrustStoreLoader.java (hunk 1 above)
Comment (ytvnr) [reply gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-core/src/main/java/io/gravitee/gateway/security/core/SubscriptionTrustStoreLoader.java]: I did not wanted to interupt the process.
digest will not be populated and a real subscription will not match for this particular loader. I guess it should be ok

Comment (ytvnr) [reply gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-core/src/main/java/io/gravitee/gateway/security/core/SubscriptionTrustStoreLoader.java]: @dev-marek I improved this by making this class throw an exception, and the caller do what he wants with it.
Our case will still be about ignoring the problem but at least is is done from another perspective

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**